### PR TITLE
Use sort/filter query to get add Work title & id to requests listing

### PR DESCRIPTION
**Pull request by kenoir**
Currently the Requests API provides only Item titles as a means to identify a request/hold. Item titles by themselves are insufficient for users to identify requested things.

This change adds the work CanonicalId and the work title of a matching work to the API response (note: items can appear on multiple works, we arbitrarily pick the one with the lowest valued sierra identifier). 

**Note for anyone picking this up**: It may be necessary to update the [API gateway description of the expected requests output](https://github.com/wellcomecollection/identity/blob/main/models/item-request-list.json#L2).

For https://github.com/wellcomecollection/catalogue-api/pull/241

### CODE REVIEW

File: requests/src/main/scala/weco/api/requests/Main.scala [hunk 1]
```diff
@@ -4,7 +4,7 @@ import akka.actor.ActorSystem
 import com.typesafe.config.Config
 import weco.Tracing
 import weco.api.requests.services.{
-  ElasticItemLookup,
```
Comment (kenoir) [inline requests/src/main/scala/weco/api/requests/Main.scala:7]: This change ditches the memory version of ItemLookup - implementing a version that worked with the `bySourceIdentifier` function when querying and sorting on works became unwieldy.

File: requests/src/main/scala/weco/api/requests/services/ItemLookup.scala [hunk 2]
```diff
@@ -1,21 +1,146 @@
...
+                    workId = workId,
+                    workTitle = title,
+                    item = item.asInstanceOf[Item[IdState.Identified]]
+                  )
+              }
+              .toList match {
+              // We can return multiple items from multiple works
+              // The sortBy above returns us the lowest bibId work association
+              case List(item) => Right(item)
+              case item :: _  => Right(item)
```
Comment (jamieparkinson) [inline requests/src/main/scala/weco/api/requests/services/ItemLookup.scala:128]: Are these cases not the same? ie I think `List(item) == item :: Nil`

**Review by jamieparkinson — APPROVED** [review]
Main thing is that I think a general comment (probably on `ItemLookup`) putting this into context would be helpful - in particular, covering all the context of why we're picking the first results when we might have several

File: requests/src/main/scala/weco/api/requests/services/ItemLookup.scala (hunk 2 above)
Comment (alexwlchan) [reply requests/src/main/scala/weco/api/requests/services/ItemLookup.scala:128]: I think this case is for the list that has more than one item.

**Comment by alexwlchan** [comment]
I'll pick up adding the comment to this; Robert and I discussed it last week.

File: requests/src/main/scala/weco/api/requests/services/ItemLookup.scala (hunk 2 above)
Comment (jamieparkinson) [reply requests/src/main/scala/weco/api/requests/services/ItemLookup.scala:128]: Right, so I think we just need the 2nd line (which will perform the work of the first too)? ie `case item :: _  => Right(item)`

Comment (alexwlchan) [reply requests/src/main/scala/weco/api/requests/services/ItemLookup.scala:128]: Yeah, I see what you mean. I'll change this.

Comment (jamieparkinson) [reply requests/src/main/scala/weco/api/requests/services/ItemLookup.scala:128]: "take the first item of a non-empty list"

**Comment by alexwlchan** [comment]
@jamieparkinson I've added some more comments to this code – mind taking a look to see if the explanation makes sense?

**Review by jamieparkinson — APPROVED** [review]
Very clear, nice

**Pull request merged**